client/src/api/combatSlice.ts

```diff
@@ -14,27 +15,33 @@ export const combatSlice = createSlice({
   name: "combat",
   initialState,
   reducers: {
-    setCombat: (state, action: PayloadAction<Combat[]>) => {
```

Comment:
I'm surprised to find that this wasn't used. Did we previously fail to end combats between turns? I see you do here. Is the title of the PR wrong (refactor means no behavior change), or am I misreading the current code? Only asking because I need to make sure the commit message is accurate when I merge.